doc/secure_boot/hld_secure_boot.md

```diff
@@ -0,0 +1,345 @@
+	* 1.6. [Architecture Design](#ArchitectureDesign)
+	* 1.7. [High-Level Design](#High-LevelDesign)
+		* 1.7.1. [Flow diagram](#Flowdiagram)
+		* 1.7.2. [Module Elements Breakdown](#ModuleElementsBreakdown)
+		* 1.7.3. [Sign Flow diagram description:](#SignFlowdiagramdescription:)
+		* 1.7.4. [Runtime SB Verification FLOW:](#RuntimeSBVerificationFLOW:)
+	* 1.8. [SAI API](#SAIAPI)
+	* 1.9. [Configuration and management](#Configurationandmanagement)
+		* 1.9.1. [CLI/YANG model Enhancements](#CLIYANGmodelEnhancements)
+		* 1.9.2. [Config DB Enhancements](#ConfigDBEnhancements)
+	* 1.10. [Warmboot and Fastboot Design Impact](#WarmbootandFastbootDesignImpact)
+	* 1.11. [Restrictions/Limitations](#RestrictionsLimitations)
+	* 1.12. [Testing Requirements/Design](#TestingRequirementsDesign)
+		* 1.12.1. [Unit Test cases](#UnitTestcases)
+		* 1.12.2. [System Test cases](#SystemTestcases)
+	* 1.13. [Open/Action items - if any](#OpenActionitems-ifany)
+
+<!-- vscode-markdown-toc-config
+	numbering=true
+	autoSave=true
+	/vscode-markdown-toc-config -->
+<!-- /vscode-markdown-toc -->
+
+###  1.1. <a name='Revision'></a>Revision
+|  Rev  |  Date   |      Author      | Change Description |
+| :---: | :-----: | :--------------: | ------------------ |
+|  0.1  | 06/2022 |   David Pilnik   | Phase 1 Design     |
+###  1.2. <a name='Scope'></a>Scope
+
+This secure boot HLD doc described the requirements, architecture, and configuration details of the secure boot feature in switches SONiC OS based.
+
+###  1.3. <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations
+	SB - Secure Boot
+	KO - Kernel Objects
+
+
+###  1.4. <a name='Overview'></a>Overview
+
+Secure Boot is a feature found in the startup software for your computer that's designed to ensure your computer starts safely and securely by preventing unauthorized software like malware from taking control of your Device at boot-up.
+
+The entire point is the "chain of trust" it creates. If I have a piece of software that I need to make sure that nothing malicious can intercept what I am doing and put their own code in I need to trust the program launching my program.
+
+What Secure boot provides is an anchor for that first "trusted program". It allows the hardware on the computer to assert "No one has modified this bootloader and it will behave exactly like the original programmers programed". The boot-loader can then check "No one has modified this OS and it will behave exactly like the original programmers programmed".(Chain of trust described below)
+
+
+###  1.5. <a name='Requirements'></a>Requirements
+
+We require to enable secure boot in SONiC OS. The feature has a signing process and a verification process. Both processes are required to be supported from the scenarios such as development and production.
+
+Assumption that Secure boot should be supported by UEFI & CPU arch and Onie OS installed in the Switch device.
+
+Enable/disable the feature in runtime:
```

Comment:
How does network admin know what devices enabled secureBoot or not? Is there a way SONiC can provide CLI for it?

Reply:
Its possible to check secure boot enable/disable status with mokutil or uefi tools from linux shell.
In SONiC currently there is no way, but, we think it is an acceptable request, so we will add a SONiC CLI command to support it.